### DISCUSSION TITLE
Remove commas when multiple warnings

### DISCUSSION THREAD

**Discussion by oskar-gmerek** [colinhacks/zod#1088] [Q&A] [answered]
Hi all,

I have question if it is possible to remove commas in easy way from message in validation when a message is an array.

`const test = ['x', 'y', 'z']`

```
    .superRefine((data, ctx) => {
      if (data.testfield >= something && data.testfield !== otherthing) {
        ctx.addIssue({
          code: z.ZodIssueCode.custom,
          message: `${test}`,
          path: ['testfield'],
        })
      }
```

`The output is: x,y,z`

I'm using more than one library for messages and all of them have translations with dot at the end so output looks bad, like that:

Unwanted output: `Some warning., Some other warning., And the 3rd warning.`

This commas behind dots looks very poor.

Wanted output: `Some warning. Some other warning. And the 3rd warning.` (dots coming from 3rd parties libraries)

It will be great if there is an easy way to sort it out as otherwise i will need to change all translations for a few languages.

**Thank you in advance!**

**Comment by scotttrinh** [comment, marked as answer]
This is more of a JavaScript question than a Zod one, but for your specific use case this should work:

```ts
    .superRefine((data, ctx) => {
      if (data.testfield >= something && data.testfield !== otherthing) {
        ctx.addIssue({
          code: z.ZodIssueCode.custom,
          message: test.join(" "),
          path: ['testfield'],
        })
      }
```

**Reply by oskar-gmerek** [reply]
Thank you very much @scotttrinh 



> This is more of a JavaScript question than a Zod one, but for your specific use case this should work

All the more thank you for helping

**Reply by scotttrinh** [reply]
No problem! We're always happy to help if we can. Best of luck.
